### ISSUE TITLE
Improve UX/UI when a user enters invalid Conversions Tracking ID in Ads module

### ISSUE THREAD

**Issue by wpdarren** [google/site-kit-wp#8475]
## Bug Description

When setting up the Ads Module, a user can currently enter an invalid `Conversions Tracking ID` and still complete the set up of the module.  When the user enters an invalid ID we should make the Complete setup CTA as disabled until they enter a valid one. This issue also applies when editing the `Conversions Tracking ID` in Ads settings, so we will need to ensure that the same applies here too. 

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWswQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b28246e8e009f8043d246462e04d8305ecd51386/image.png" alt="image.png" />

## Steps to reproduce

1. Ensure that the  `adsModule` feature flag is enabled. 
2. Go to setup the Ads module and in the `Conversions Tracking ID` enter an invalid ID. 
3. Notice that the `Complete setup` CTA is active and the input can be saved. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

When setting up the Ads module, the following criteria should be satisfied:
* If a user enters an invalid Conversion Tracking ID value within the settings field, the "Complete setup" CTA button should be disabled, disallowing module set up when an invalid Conversion ID value is entered.

## Implementation Brief

* [ ] 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by bethanylang** [comment]
> This issue also applies when editing the Conversions Tracking ID in Ads settings, so we will need to ensure that the same applies here too.

@aaemnnosttv has created #8508 to address this more broadly in settings, as it's a regression from #7533.

**Comment by 10upsimon** [comment]
It looks to me like we can close this in favor of https://github.com/google/site-kit-wp/issues/8508 which states in the AC the following:

> Module settings should have the button to save changes disabled while the input for any of its settings is invalid


And therefore should apply to Ads as well.

**Comment by bethanylang** [comment]
@10upsimon Confirming per [Slack](https://10up.slack.com/archives/CBKKQEBR9/p1712825558230139) that this issue only covers the Ads module field while #8508 is for the broader issue with fields for all modules, so keeping both open.  :)

**Comment by eugene-manuilov** [comment]
Thanks, @10upsimon. AC mostly looks good to me, but since we have #8508 in EB now, we don't need to change how settings work in this ticket and update only the setup part. Could you please remove the part that says that refers to the editing settings?

**Comment by eugene-manuilov** [comment]
AC ✔️

**Comment by benbowler** [comment]
@eugene-manuilov @10upsimon @bethanylang I've actually already implemented everything in this AC in #8508 as I misunderstood the IB over there. I will attach that PR here and create an IB. Once the IB is approved I can make any required changes to the PR.

**Comment by benbowler** [comment]
@eugene-manuilov @10upsimon @bethanylang After looking into #8505 and addressing the AC correctly there, the AC of this ticket has already been covered by the new PR #8545.

This is because the field, form submit buttons and validation logic are shared between the setup and edit screens so in order to complete this AC point:...

 > - [Module settings should have the button to save changes disabled while the input for any of its settings is invalid](https://github.com/google/site-kit-wp/issues/8508)

...the settings edit screen in the Ads module was also updated to behave the same way with no additional work.

**That means that this ticket can be closed in favour of #8505.**

**Comment by benbowler** [comment]
#8508 is now merged, I suggest this ticket is still QA'd to confirm that the AC has been addressed in that merged ticket. CC @bethanylang @wpdarren

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:** 

- If a user enters an invalid `Conversion Tracking ID` value within the settings field, the `Complete setup` CTA button is disabled, disallowing module set up when an invalid Conversion ID value is entered.

https://github.com/google/site-kit-wp/assets/73545194/512613db-627c-408d-8410-0603a87dc9d2

**Comment by aaemnnosttv** [comment]
Thanks @benbowler @wpdarren – closing this one out since no additional work was needed for it after all 👍
